File: Others/LeetCode/Interview150/26-remove_duplicates_from_sorted_array.ts
function removeDuplicates(nums: number[]): number {
  if (nums.length === 0){
    return 0;
  }
  let write_index = 1;
  for (let i= 1; i < nums.length; i++){
    // console.log('Index',i,'Value',value, 'Last array val', new_nums[write_index - 1]);
    if (nums[i] != nums[write_index - 1]){
      // console.log('Adding ', value);
      nums[write_index] = nums[i];
      write_index += 1;
    }
  }
  // console.log('Pre assign', new_nums);
  return write_index;
};
//
let nums: number[] = [-1,0,0,0,0,3,3]
const r = removeDuplicates(nums)
console.log('Result: ', r, nums.slice(0, r));

File: Others/LeetCode/Interview150/26-remove_duplicates_from_sorted_array.py
class Solution(object):
    def removeDuplicates(self, nums: list[int]):
        """
        :type nums: List[int]
        :rtype: int
        """
        processed_nums : dict[int, int] = {}
        for i in nums:
            if processed_nums.get(i) == None:
                processed_nums[i] = 1;
            else:
                continue
        nums[:] = list(processed_nums.keys())
        # print('Nums:',nums)
        return nums.__len__()


if __name__ == '__main__':
    test = Solution()
    nums = [-1,0,0,0,0,3,3]
    result = test.removeDuplicates(nums)
    print("Result: ", result, nums)

